src/modules/main/components/NewestContainer.tsx

```diff
@@ -24,7 +29,7 @@ const NewestContainer: React.FC = () => {
   });
 
   useEffect(() => {
-    setNewest();
+    setNewestInitial();
```

Comment:
additional thunk for initial load is redundant, we can check if there are posts right here :)
```
    if (newestPosts.length === 0) {
      setNewest();
    }
```

Reply:
On one of the syncs, we have discussed using the separate thunk for the initial load.

Checking length in the actual component was the first solution but was rejected.